jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -480,6 +480,11 @@ private Runnable onFailure(Throwable x, boolean remote)
                 };
 
                 // Serialize all the error actions.
+
+//                Case 2
+//                if (invokeOnContentAvailable != null || invokeWriteFailure != null)
+//                    invokeOnFailureListeners = null;
+
```

Comment:
Our documentation explicitly says that we will call the listener even if there is are demand and write callbacks.

Perhaps we should allow the listener to be a biconsumer, and pass in if the application has been notified in some other way?   But is option 3 enough to fix this?